src/ORM/Search/SearchContext.php

```diff
@@ -176,40 +205,117 @@ public function getQuery($searchParams, $sort = false, $limit = false, $existing
             $query = $query->limit($limit);
         }
 
-        /** @var DataList $query */
-        $query = $query->sort($sort);
-        $this->setSearchParams($searchParams);
+        return $query->sort($sort);
+    }
 
-        $modelObj = Injector::inst()->create($this->modelClass);
-        $searchableFields = $modelObj->searchableFields();
-        foreach ($this->searchParams as $key => $value) {
-            $key = str_replace('__', '.', $key ?? '');
-            if ($filter = $this->getFilter($key)) {
-                $filter->setModel($this->modelClass);
-                $filter->setValue($value);
-                if (!$filter->isEmpty()) {
-                    if (isset($searchableFields[$key]['match_any'])) {
-                        $searchFields = $searchableFields[$key]['match_any'];
-                        $filterClass = get_class($filter);
-                        $modifiers = $filter->getModifiers();
-                        $query = $query->alterDataQuery(function (DataQuery $dataQuery) use ($searchFields, $filterClass, $modifiers, $value) {
-                            $subGroup = $dataQuery->disjunctiveGroup();
-                            foreach ($searchFields as $matchField) {
-                                /** @var SearchFilter $filterClass */
-                                $filter = new $filterClass($matchField, $value, $modifiers);
-                                $filter->apply($subGroup);
-                            }
-                        });
-                    } else {
-                        $query = $query->alterDataQuery([$filter, 'apply']);
+    /**
+     * Use the global primary search for searching across multiple fields
+     *
+     * @param string|array $searchPhrase
+     */
+    protected function primaryFieldSearch(DataList $query, array $searchableFields, $searchPhrase): DataList
+    {
+        $context = $this;
+        $modelClass = $this->modelClass;
+        return $query->alterDataQuery(function (DataQuery $dataQuery) use ($context, $modelClass, $searchableFields, $searchPhrase) {
```

Comment:
```suggestion
        return $query->alterDataQuery(function (DataQuery $dataQuery) use ($this, $modelClass, $searchableFields, $searchPhrase) {
```
I'm not 100% sure if it's necessary to pass `$this` in to anonymous functions or if it's already available

Reply:
I don't know what you mean by that?

Reply:
removed - turns out `$this` is already bound to the anonymous function automagically.